solution/backend/common/fields.py

```diff
@@ -74,6 +74,16 @@ def naturalize_int_match(match):
         return string
 
 
+#  Allows you to pass in an array of fields instead of just one.  It checks to see if the other fields are blank
+#  To determine sorting, otherwise will do the same as natural sort field.
+class CombinedNaturalSort(NaturalSortField):
+    def pre_save(self, model_instance, add):
+        for field in self.for_field:
```

Comment:
we return the first field that has anything in it for our ordering.  So if shortname is empty then it will go to the next one which is abbreviation. The higherarchy is defined in the model.